server/bin/pbench-server-prep-shim-002.py

```
@@ -0,0 +1,307 @@
+
+
+_NAME_ = "pbench-server-prep-shim-002"
+
+
+class Results(object):
+    def __init__(
+        self, nstatus="", ntotal=0, ntbs=0, nquarantined=0, ndups=0, nerrs=0,
+    ):
+        self.nstatus = nstatus
+        self.ntotal = ntotal
+        self.ntbs = ntbs
+        self.nquarantined = nquarantined
+        self.ndups = ndups
+        self.nerrs = nerrs
+
+
+def fetch_config_val(config, errorfile):
+
+    qdir = config.get("pbench-server", "pbench-quarantine-dir")
+    if not qdir:
+        errorfile.write("Failed: getconf.py pbench-quarantine-dir pbench-server\n")
+        return None, None
+
+    qdir = Path(qdir).resolve()
+    if not qdir.is_dir():
+        errorfile.write(f"Failed: {qdir} does not exist, or is not a directory\n")
+        return None, None
+
+    # we are explicitly handling version-002 data in this shim
+    receive_dir_prefix = config.get("pbench-server", "pbench-receive-dir-prefix")
+    if not receive_dir_prefix:
+        errorfile.write("Failed: getconf.py pbench-receive-dir-prefix pbench-server\n")
+        return None, None
+
+    receive_dir = Path(f"{receive_dir_prefix}-002").resolve()
+    if not receive_dir.is_dir():
+        errorfile.write(
+            f"Failed: {receive_dir} does not exist, or is not a directory\n"
+        )
+        return None, None
+
+    return (qdir, receive_dir)
+
+
+def qdirs_check(qdir_val, qdir, logger):
+    try:
+        os.makedirs(qdir)
+    except FileExistsError:
+        # directory already exists, ignore
+        pass
+    except Exception:
+        logger.exception(
+            "os.mkdir: Unable to create {} destination directory: {}", qdir_val, qdir,
+        )
+        return None
+    return qdir
+
+
+def md5_check(tb, tbmd5, logger):
+    # read the md5sum from md5 file
+    try:
+        with tbmd5.open() as f:
+            archive_md5_hex_value = f.readline().split(" ")[0]
+    except Exception:
+        logger.exception("Quarantine: Could not read {}", tbmd5)
```

If we get an exception, the `archive_md5_hex_value` variable will be undefined when we return at line 96.

Could we raise an exception that will be caught and handled at the time `md5_check` is called?

---

An alternative to raising an exception is to keep logging it, but make sure we set `archive_md5_hex_value` to `None` so the value is defined when returned at line 96.